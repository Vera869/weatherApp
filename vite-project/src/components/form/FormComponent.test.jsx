import {render} from "@testing-library/react";
import Form from "./FormComponent";
import { expect, it, vitest } from "vitest";
import { useSelector } from "react-redux";

vitest.mock("react-redux");

it("Отображение формы", () => {
  useSelector.mockReturnValue("");
  const component = render(<Form/>);
  expect(component).toMatchSnapshot();
});