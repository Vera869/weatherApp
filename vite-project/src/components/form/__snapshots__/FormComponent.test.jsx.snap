// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Отображение формы 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="_container_940aab"
      >
        <h2
          class="_header_940aab"
        >
          Выберите город
        </h2>
        <input
          class="_Input_940aab"
          name="city"
          placeholder="Город"
          type="text"
          value=""
        />
        <p
          class="_Input_errors_940aab"
        />
        <button
          class="_Button_940aab"
        >
          Узнать погоду
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="_container_940aab"
    >
      <h2
        class="_header_940aab"
      >
        Выберите город
      </h2>
      <input
        class="_Input_940aab"
        name="city"
        placeholder="Город"
        type="text"
        value=""
      />
      <p
        class="_Input_errors_940aab"
      />
      <button
        class="_Button_940aab"
      >
        Узнать погоду
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
